src/language/branches.md

```diff
@@ -0,0 +1,279 @@
+  print('Was coordinate array $x,$y');
+} else {
+  throw FormatException('Invalid coordinates.');
+}
+```
+
+If-case is a simple way to match and destructure against a
+_single_ pattern. To test a value against _multiple_ patterns, use [switch](#switch).
+
+## Switch
+
+Switch statements evaluate a value expression against a series of cases.
+Each case clause is a [pattern][] followed by a series of statements in the case
+body to execute if the value matches that pattern. 
+
+Non-empty `case` clauses implicitly jump to the end of the switch after completion.
+Other valid ways to end a non-empty `case` clause are a [`continue`][break],
+[`throw`][], or [`return`][] statement.
+
+Use a `default` or wildcard `_` clause to execute code when no `case` clause matches:
+
+```dart
+var command = 'OPEN';
+switch (command) {
+  case 'CLOSED':
+    executeClosed();
+  case 'PENDING':
+    executePending();
+  case 'APPROVED':
+    executeApproved();
+  case 'DENIED':
+    executeDenied();
+  case 'OPEN':
+    executeOpen();
+  default:
+    executeUnknown();
+}
+```
+
+Empty cases fall through to the next case. For non-sequential fall-through,
+you can use a [`continue` statement][break] and a label:
+
+```dart
+switch (command) {
+  case 'OPEN':
+    executeOpen();
+    continue newCase;     // Continues executing at the newCase label.
+  
+  case 'DENIED':          // Empty case falls through.
+  case 'CLOSED':
+    executeClosed();      // Runs for both DENIED and CLOSED,
+  
+  newCase:
+  case 'PENDING':
+    executeNowClosed();   // Runs for both OPEN and PENDING.
+}
+```
+For an empty case that does not fall through, use [`break`][break] for its body.
+
+### Case patterns
+
+The pattern following a case, or case pattern, allows control flow to
+either match and destructure the object being switched on, or continue execution
+if the object doesn't match. Case patterns can be any kind of [pattern][]:
+
+- **Constant patterns**: the object's value has to be equivalent to match.
+  ```dart
+  switch (number) {
+    case 1:           // Matches if number == 1
+      print("one");     
+  }
+  ```
+- **List**, **map**, or **record patterns**: the elements are destructured if matched.
+  ```dart
+  switch (obj) {
+    case [a, b]:                  // Matches if obj is a list with two elements
+      print("$a and $b match");    
+  }
+  ```  
+- **Variable patterns**: the matched value binds to a new variable.
+  ```dart
+  switch ((1, 2)) {
+    case (var a, var b): ...     // Matches if 1 and 2 can be assigned to a and b
+  }
+  ```  
+- **Logical** or **relational patterns**: the object matches if the comparison to
+a constant returns `true`.
+  ```dart
+  switch (code) {
+    case >= first && <= last: ...    
+    // Matches if the value of code is between the constant values of first and last
+  }
+  ```  
+- **Null check** or **assert patterns**: the match is first on whether the object is not null, then on the value.
```

Comment:
I would split null-check and null-assert patterns into two separate sections since they behave pretty differently on null. (Also, the latter are called "null assert" patterns not just "assert" patterns since they're specific to handling null.)

Also, the wording is pretty strange. Maybe: If the object is `null`, then doesn't match. Otherwise, matches the object against the inner pattern.

Reply:
Moved this section to the types reference page where we addressed this, resolving!